pykwalify/core.py

```diff
@@ -653,6 +659,40 @@ def _check_int_timestamp_boundaries(timestamp):
                 timestamp=timestamp_value,
             ))
 
+
+    def _validate_scalar_date(self, date_value, date_format, path):
+        log.debug(u"Validate date : %(value)s : %(format)s : %(path)s" % {'value': date_value,
+                                                                          'format':date_format,
+                                                                          'path': path})
+
+        if isinstance(date_value, str):
+            if date_format is None:
+                raise CoreError(u'If date does not match yaml date specification a format must be specify in grammar')
+            import time
```

Comment:
Local imports only valid in case you have circular imports. Move this to top of file.